hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/client/FlinkTaskContextSupplier.java

```diff
@@ -62,4 +62,9 @@ public Option<String> getProperty(EngineProperty prop) {
     return Option.empty();
   }
 
+  @Override
+  public Supplier<Integer> getAttemptNumberSupplier() {
+    return () -> -1;
```

Comment:
In Flink we already have `getAttemptIdSupplier` which serves as the same purpose, what's the usage of Spark then? should we use `getAttemptIdSupplier` instead ?

Reply:
as of now, I have disabled it for flink. can you test it out and fix it.

here is the situation w/ spark. 

Lets say we want to spin up 10 tasks for a stage. 
in first attempt, 
each task will be assigned numbers from 1 to 10 for attemptId. but for attempNumber, it will be 0. 
and when a subset of tasks are retried, new attemptIds will be 11, 12 etc. but attemptNumber will be 1.  

This is how it works in spark. I am not very sure on flink. Anyways, w/ latest commit, we are avoiding writing the block identifier flink.

Reply:
So, in summary, attemptId supplier is not the right one to use atleast in spark. It has to be attempt number.

Reply:
Does spark retry always takes the same data set? Is is possible one retried task goes to other executor/container and takes a different input dataset there?

Reply:
wrt log files, it should not, bcoz, only updates go to log files. 
incase of bucket Index, anyways, record key to fileId is hash based and we should be good.

Reply:
> only updates go to log files.

Only true for spark, so you are fixing a bug dependent on Spark write christeristic.

Reply:
yes. we have disabled it for flink as of now. In java, anyways, MOR is not fully functional from what I know. but I can open to disabling it for java as well. mainly its an issue for ExpressionPayload and any other custom payloads. most of the other payloads are idempotent even if there are duplicate log blocks.

Reply:
Let's make the Flink impl right first by using `this.flinkRuntimeContext.getAttemptNumber()`

Reply:
sure, will sync up w/ you directly to better understand this. 
https://issues.apache.org/jira/browse/HUDI-6844